### ISSUE TITLE
Implement Spiking Neural Dynamics for Energy-Efficient Processing

### ISSUE THREAD

**Issue by DrDrewCain** [DrDrewCain/MiniMacLLM#23]
## Overview
Implement spiking neural network (SNN) dynamics for energy-efficient processing and memory compression.

## Priority
⭐⭐⭐⭐ (High Priority)

## Expected Benefits
- **30-50% energy reduction** on neuromorphic hardware
- **16x replay buffer compression** (1-bit spikes vs 16-bit floats)
- **66% faster training convergence** (from SpikeGPT paper)
- Event-driven processing with temporal precision

### 1. Core Components

**Leaky Integrate-and-Fire (LIF) Neuron**:
```python
U[t] = H[t] + β(Y[t] - (H[t-1] - U_reset))
S[t] = Θ(U[t] - U_threshold)    # Binary spike output
H[t] = U[t] · (1 - S[t])         # Reset after spike
```

**Parameters**:
- β = 0.5 (leak coefficient)
- U_threshold = 1.0
- U_reset = 0.0
- Target firing rate: 15% (85% sparsity)

**Surrogate Gradients**: Use arctangent approximation for backpropagation

### 2. Integration Strategy

**Hybrid Architecture**:
- Keep attention continuous
- Make FFN spiking (80% of compute)
- Layers 9-12: Spiking FFN

**Spiking Experience Replay**: Store spike indices (16x compression)

### 3. Implementation Files

- `src/neurobio/spiking.py`
- `src/model/spiking_ffn.py`
- `src/continual/spiking_replay.py`
- Tests

## Key Papers

- **SpikeGPT**: Zhu et al., 2023. arXiv:2302.13939
- **TIM Module**: Yao et al., 2024. IJCAI 2024
- **Surrogate Gradients**: Neftci et al., 2019. IEEE SPM
- **Neuromorphic Survey**: Roy et al., 2019. Nature 575:607-617

## Success Criteria

- LIF neurons with surrogate gradients ✓
- SpikingSwiGLU functional ✓
- 15% firing rate achieved ✓
- 16x compression demonstrated ✓
- All tests passing ✓
- 30%+ energy reduction (simulated) ✓
